src/hierarchies/updaters/fa_updater.h

```diff
@@ -0,0 +1,37 @@
+#ifndef BAYESMIX_HIERARCHIES_UPDATERS_FA_UPDATER_H_
+#define BAYESMIX_HIERARCHIES_UPDATERS_FA_UPDATER_H_
+
+#include "abstract_updater.h"
+#include "src/hierarchies/likelihoods/fa_likelihood.h"
+#include "src/hierarchies/likelihoods/states/includes.h"
+#include "src/hierarchies/priors/fa_prior_model.h"
+#include "src/hierarchies/priors/hyperparams.h"
+#include "src/utils/proto_utils.h"
+
+class FAUpdater : public AbstractUpdater {
+ public:
+  FAUpdater() = default;
+  ~FAUpdater() = default;
+  void draw(AbstractLikelihood& like, AbstractPriorModel& prior,
+            bool update_params) override;
+
+ protected:
+  void sample_eta(State::FA& state, const Hyperparams::FA& hypers,
+                  const FALikelihood& like);
+  void sample_mu(State::FA& state, const Hyperparams::FA& hypers,
+                 const FALikelihood& like);
+  void sample_lambda(State::FA& state, const Hyperparams::FA& hypers,
+                     const FALikelihood& like);
+  void sample_psi(State::FA& state, const Hyperparams::FA& hypers,
+                  const FALikelihood& like);
+  // void sample_eta(State::FA & state, const Hyperparams::FA & hypers, const
+  // Eigen::MatrixXd * dataset_ptr, const std::set<int> & cluster_data_idx);
+  // void sample_mu(State::FA & state, const Hyperparams::FA & hypers, const
+  // Eigen::VectorXd & data_sum); void sample_lambda(State::FA & state, const
+  // Hyperparams::FA & hypers, const Eigen::MatrixXd * dataset_ptr, const
+  // std::set<int> & cluster_data_idx, size_t dim); void sample_psi(State::FA &
+  // state, const Hyperparams::FA & hypers, const Eigen::MatrixXd *
+  // dataset_ptr, const std::set<int> & cluster_data_idx, size_t dim);
```

Comment:
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.